utils/alexa/ssl_tools.py

```diff
@@ -0,0 +1,201 @@
+    """
+    parsed = urlsplit(url)
+
+    scheme: str = parsed.scheme
+    netloc: str = parsed.netloc
+    path: str = parsed.path
+
+    try:
+        port = parsed.port
+    except ValueError:
+        port = None
+
+    result = (scheme.lower() == 'https' and
+              netloc.lower().split(':')[0] == 's3.amazonaws.com' and
+              path[:10] == '/echo.api/' and
+              (port == 443 or port is None))
+
+    return result
+
+
+def extract_certs(certs_txt: str) -> List[crypto.X509]:
+    """Extracts pycrypto X509 objects from SSL certificates chain string.
+
+    Args:
+        certs_txt: SSL certificates chain string.
+
+    Returns:
+        result: List of pycrypto X509 objects.
+    """
+    pattern = r'-----BEGIN CERTIFICATE-----.+?-----END CERTIFICATE-----'
+    certs_txt = re.findall(pattern, certs_txt, flags=re.DOTALL)
+    certs = [crypto.load_certificate(crypto.FILETYPE_PEM, cert_txt) for cert_txt in certs_txt]
+    return certs
+
+
+def verify_sans(amazon_cert: crypto.X509) -> bool:
+    """Verifies Subject Alternative Names (SANs) for Amazon certificate.
+
+    Args:
+        amazon_cert: Pycrypto X509 Amazon certificate.
+
+    Returns:
+        result: True if verification was successful, False if not.
+    """
+    cert_extentions = [amazon_cert.get_extension(i) for i in range(amazon_cert.get_extension_count())]
+    subject_alt_names = ''
+
+    for extention in cert_extentions:
+        if 'subjectAltName' in str(extention.get_short_name()):
+            subject_alt_names = extention.__str__()
+            break
+
+    result = 'echo-api.amazon.com' in subject_alt_names
+
+    return result
+
+
+def verify_certs_chain(certs_chain: List[crypto.X509], amazon_cert: crypto.X509) -> bool:
+    """Verifies if Amazon and additional certificates creates chain of trust to a root CA.
+
+    Args:
+        certs_chain: List of pycrypto X509 intermediate certificates from signature chain URL.
+        amazon_cert: Pycrypto X509 Amazon certificate.
+
+    Returns:
+        result: True if verification was successful, False if not.
+    """
+    store = crypto.X509Store()
+
+    # add certificates from Amazon provided certs chain
+    for cert in certs_chain:
+        store.add_cert(cert)
+
+    # add CA certificates
+    default_verify_paths = ssl.get_default_verify_paths()
+
+    default_verify_file = default_verify_paths.cafile
+    default_verify_file = Path(default_verify_file).resolve() if default_verify_file else None
+
+    default_verify_path = default_verify_paths.capath
+    default_verify_path = Path(default_verify_path).resolve() if default_verify_path else None
+
+    ca_files = [ca_file for ca_file in default_verify_path.iterdir()] if default_verify_path else []
+    if default_verify_file:
+        ca_files.append(default_verify_file)
+
+    if ca_files:
+        for ca_file in ca_files:
+            ca_file: Path = ca_file
+            if ca_file.is_file():
+                with ca_file.open('r') as crt_f:
```

Comment:
You should add an encoding, if you're reading this file as text and not as bytes